.github/workflows/tests.yml

```diff
@@ -0,0 +1,33 @@
+name: Tests for minbpe
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+    - "minbpe/**.py"
+    - "tests/**.py"
+  push:
+    branches:
+      - master
+    paths:
+    - "minbpe/**.py"
+    - "tests/**.py"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+      - name: Install dependencies
```

Comment:
Happy to check this against setup cache too but wanted to keep it minimal and simple.